astropy/utils/decorators.py

```diff
@@ -607,6 +619,29 @@ def wrapper(*args, **kwargs):
     return decorator
 
 
+def deprecated_keywords(*kws, since):
+    """Deprecate calling one or more arguments as keywords.
+
+    Parameters
+    ----------
+    *kws: str
+        Names of the arguments that will become positional-only.
+
+    since : str or number or sequence of str or number
+        The release at which the old argument became deprecated.
+    """
+    return partial(_depr_kws, kws=kws, since=since)
+
+
+def _depr_kws(func: F, /, kws: tuple[str, ...], since: str) -> F:
+    if not callable(func):
+        raise TypeError("not a callable")
```

Comment:
A quick test for this would be great.

Reply:
actually this is vestigial of `positional_defaults`, which for some reason had more complexity in that the equivalent function was overloaded. I simplified it so I might as well drop this branch too.